qt/widgets/common/src/AlgorithmProgress/AlgorithmProgressDialogWidget.cpp

```diff
@@ -0,0 +1,83 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressDialogWidget.h"
+#include "MantidQtWidgets/Common/AlgorithmProgress/AlgorithmProgressModel.h"
+
+#include "MantidQtWidgets/Common/MantidWidget.h"
+#include <QHeaderView>
+#include <QIcon>
+#include <QProgressBar>
+#include <QPushButton>
+#include <QStringList>
+#include <QTreeWidget>
+#include <QVBoxLayout>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+AlgorithmProgressDialogWidget::AlgorithmProgressDialogWidget(
+    QWidget *parent, AlgorithmProgressModel &model)
+    : QDialog(parent),
+      m_presenter{std::make_unique<AlgorithmProgressDialogPresenter>(
+          parent, this, model)},
+      m_tree{new QTreeWidget(this)} {
+  setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
The issue I had here was that then I have to hook the parent to the widget's close event in order to properly delete it. 

I got around that by adding `setAttribute(Qt::WA_DeleteOnClose)` to the child, and storing the pointer in a [`QPointer`](https://doc.qt.io/qt-5/qpointer.html#details) which guarantees that it will be set to `nullptr` when the child is deleted (which it does itself when closed).